enclave-tls/cmake/SGXSSL.cmake

```diff
@@ -0,0 +1,91 @@
+# Project name
+if(SGX)
+    project(intel-sgx-ssl)
+endif()
+
+include(ExternalProject)
+include(FindSGXSSL)
+
+# Patch
+set(_patch_script "${CMAKE_CURRENT_SOURCE_DIR}/patch_script.sh")
+file(WRITE "${_patch_script}"
+"#!/bin/sh
+cd ${INTEL_SGXSSL_SRC}/
+patch -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/patch/0001-add-ssl-library-enclave-support.patch
+")
+
+set(_patch_cmake "${CMAKE_CURRENT_SOURCE_DIR}/patch.cmake")
+file(WRITE "${_patch_cmake}"
+        "execute_process(COMMAND sh ${_patch_script} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})"
+)
+
+# Configure
+set(_configure_script "${CMAKE_CURRENT_SOURCE_DIR}/download_script.sh")
+file(WRITE "${_configure_script}"
+"#!/bin/sh
+if [ ! -f ${OPENSSL_DIR}/openssl-1.1.1*.tar.gz ]; then
+       wget --no-check-certificate https://www.openssl.org/source/openssl-1.1.1k.tar.gz -P ${OPENSSL_DIR};
```

Comment:
Whether need to integrate this openssl lib as a submodule into the whole project. After that, user can use git submodule to download this package automatically.

Reply:
openssl is needed as tarball not source code.